packages/flutter/lib/src/widgets/automatic_keep_alive.dart

```diff
@@ -319,19 +319,6 @@ class KeepAliveNotification extends Notification {
 /// consider using [AutomaticKeepAliveClientMixin], which uses a
 /// [KeepAliveHandle] internally.
 class KeepAliveHandle extends ChangeNotifier {
-  /// Trigger the listeners to indicate that the widget
-  /// no longer needs to be kept alive.
-  ///
-  /// This method does not call [dispose]. When the handle is not needed
-  /// anymore, it must be [dispose]d regardless of whether notifying listeners.
-  @Deprecated(
-    'Use dispose instead. '
```

Comment:
If we've told users they should replace this with dispose, we should have a dart fix for it. The original author should be able to confirm. I think it was @dnfield?

Reply:
KeepAliveHandle.release and KeepAliveHandle.dispose seem to have different semantics as the latter also releases all resources, and typically shouldn't be called a second time (see the example below).  The closer alternative seems to be `notifyListeners` but it's protected.

```dart
handler.release();
handler.dispose();
```
would be migrated to:
```dart
handler.dispose();
handler.dispose();
```
and the second dispose call is redundant and it's probably going to assert in debug mode.

Reply:
I think the motivation to deprecate was that folks were only calling release, and not calling dispose, but this one might be ok without a fix. We typically provide a fix in every case we can. Even if it is not a perfect fix, it helps users get to the final result.